youtube2zim/scraper.py

```diff
@@ -550,24 +572,71 @@ def get_slot():
 
         return overall_succeeded, overall_failed
 
-    @staticmethod
-    def download_video_files_batch(options, videos_ids):
+    def download_s3_cache(self, s3_key, local_video_path):
+        """returns True if successfully downloaded cache, otherwise False"""
+        try:
+            if self.use_any_optimized_version:
+                if not self.s3_storage.has_object(s3_key, self.bucket_name):
+                    return False
+            else:
+                if not self.s3_storage.has_object_matching_meta(
+                    s3_key,
+                    tag="encoding_version",
+                    value=ENCODER_VERSION,
+                    bucket_name=self.bucket_name,
+                ):
+                    return False
+            os.makedirs(os.path.dirname(local_video_path), exist_ok=True)
+            self.s3_storage.download_file(s3_key, local_video_path)
+        except Exception as ex:
+            logger.error(f"{self.bucket_name}/{s3_key} download failed with {str(ex)}")
+            return False
+        logger.info(
+            f"cache ({self.bucket_name}/{s3_key}) downloaded {local_video_path}"
+        )
+        return True
 
+    def upload_s3_cache(self, s3_key, local_video_path):
+        """returns True if video uploaded to S3 for cache otherwise None"""
+        try:
+            self.s3_storage.upload_file(
+                local_video_path, s3_key, meta={"encoding_version": ENCODER_VERSION}
+            )
+            logger.info(f"cached video to {self.bucket_name} bucket ({s3_key})")
+            return True
+        except Exception as ex:
+            logger.error(f"{self.bucket_name}/{s3_key} upload failed with {str(ex)}")
+
+    def download_video_files_batch(self, options, videos_ids):
         succeeded = []
         failed = []
-        with youtube_dl.YoutubeDL(options) as ydl:
-            for video_id in videos_ids:
-                try:
+        for video_id in videos_ids:
+            video_location = options["y2z_videos_dir"].joinpath(video_id)
+            # Reset the skip_download argument
```

Comment:
asked you to add a comment for this (because option name is not explicit: doesn't downloads the video but downloads the rest)

Reply:
I don't understand what you mean. Though I have a comment few statements below.

Reply:
the youtube-dl option is not self explaintatory. One not familiar with it might think it doesn't download anything so please explain what this option does (skip video dl but downloads the rest)